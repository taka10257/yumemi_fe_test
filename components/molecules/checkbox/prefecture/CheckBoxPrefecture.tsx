import React, { ChangeEvent } from 'react'
import styles from './CheckBoxPrefecture.module.sass'

type CheckBoxPrefectureProps = {
  prefCode: number
  prefName: string
  onCheck(prefCode: number, prefName: string): void
  onUnCheck(prefCode: number): void
}

export default function CheckBoxPrefecture({
  prefCode,
  prefName,
  onCheck,
  onUnCheck,
}: CheckBoxPrefectureProps) {
  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (event.target.checked) {
      onCheck(prefCode, prefName)
      return
    }
    onUnCheck(prefCode)
  }

  return (
    <label className={styles.label}>
      <input type="checkbox" value={prefCode} onChange={onChange} />
      <span>{prefName}</span>
    </label>
  )
}
