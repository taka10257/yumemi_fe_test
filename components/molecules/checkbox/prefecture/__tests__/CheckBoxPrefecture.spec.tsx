import React from 'react'
import { shallow } from 'enzyme'
import CheckBoxPrefecture from '../CheckBoxPrefecture'
import { expect } from '@jest/globals'

describe('components/molecules/checkbox/prefecture/CheckBoxPrefecture', () => {
  describe('正常系', () => {
    it('label/valueの確認', () => {
      const checkbox = shallow(
        <CheckBoxPrefecture
          prefCode={1}
          prefName={'北海道'}
          onCheck={jest.fn}
          onUnCheck={jest.fn}
        />
      )
      expect(checkbox).toMatchSnapshot()
    })

    it('onCheck/onUnCheckの確認', () => {
      const prefCode = 1
      const prefName = '北海道'

      let resultPrefCode = 0
      let resultPrefName = ''

      const mockOnCheck = (_prefCode: number, _prefName: string) => {
        resultPrefCode = _prefCode
        resultPrefName = _prefName
      }
      const mockOnUnCheck = (_prefCode: number) => {
        resultPrefCode = _prefCode
        resultPrefName = ''
      }

      const checkbox = shallow(
        <CheckBoxPrefecture
          prefCode={prefCode}
          prefName={prefName}
          onCheck={mockOnCheck}
          onUnCheck={mockOnUnCheck}
        />
      )

      const input = checkbox.find('input')

      input.simulate('change', { target: { checked: true } })
      expect(resultPrefCode).toEqual(prefCode)
      expect(resultPrefName).toEqual(prefName)

      input.simulate('change', { target: { checked: false } })
      expect(resultPrefCode).toEqual(prefCode)
      expect(resultPrefName).toEqual('')
    })
  })
})
