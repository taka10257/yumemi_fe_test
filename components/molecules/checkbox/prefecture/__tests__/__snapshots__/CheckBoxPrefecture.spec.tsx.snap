// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/checkbox/prefecture/CheckBoxPrefecture 正常系 label/valueの確認 1`] = `
<label
  className="label"
>
  <input
    onChange={[Function]}
    type="checkbox"
    value={1}
  />
  <span>
    北海道
  </span>
</label>
`;
